caddy.go

```diff
@@ -867,3 +867,19 @@ var (
 	// by default if no other file is specified.
 	DefaultConfigFile = "Caddyfile"
 )
+
+// contextKey is a value for use with context.WithValue. Inspired by the
+// net/http source.
+type contextKey struct {
+	name string
+}
```

Comment:
I wonder if this belongs in the httpserver package. (My PR, #1430 creates a type called `CtxKey` so maybe we should consider merging these implementations.) I like the simplicity of it just being a string... any reason to use a struct?